plugins/microsoft-calendar/src/components/CalendarEvent.tsx

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright 2022 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import classnames from 'classnames';
+import {
+  bindPopover,
+  bindTrigger,
+  usePopupState,
+} from 'material-ui-popup-state/hooks';
+import React, { useState } from 'react';
+
+import { Link } from '@backstage/core-components';
+
+import {
+  Box,
+  Paper,
+  Popover,
+  Tooltip,
+  Typography,
+  makeStyles,
+} from '@material-ui/core';
+
+import webcamIcon from '../icons/webcam.svg';
+import { CalendarEventPopoverContent } from './CalendarEventPopoverContent';
+import { MicrosoftCalendarEvent } from '../api';
+import {
+  getOnlineMeetingLink,
+  getTimePeriod,
+  isAllDay,
+  isPassed,
+} from './util';
+
+const useStyles = makeStyles(
+  theme => ({
+    event: {
+      display: 'flex',
+      alignItems: 'center',
+      marginBottom: theme.spacing(1),
+      cursor: 'pointer',
+      paddingRight: 12,
+    },
+    declined: {
+      textDecoration: 'line-through',
+    },
+    passed: {
+      opacity: 0.6,
+      transition: 'opacity 0.15s ease-in-out',
+      '&:hover': {
+        opacity: 1,
+      },
+    },
+    link: {
+      width: 48,
+      height: 48,
+      display: 'inline-block',
+      padding: 8,
+      borderRadius: '50%',
+      '&:hover': {
+        backgroundColor: theme.palette.grey[100],
+      },
+    },
+    calendarColor: () => ({
```

Comment:
This doesn't need to be a function right?

Reply:
yes, removed